### ISSUE TITLE
DearEIS added to the Dear PyGui showcase

### ISSUE THREAD

**Issue by bandit-masked** [vyrjana/DearEIS#3]
Hey,

I just wanted to let you know that the Dear PyGui creators and moderators think DearEIS looks absolutely fantastic and that I have added it to the [Dear PyGui Showcase](https://github.com/hoffstadt/DearPyGui/wiki/Dear-PyGui-Showcase) gallery. A bit late, given that you posted about it a few months ago, but better late than never.

I have also posted the GIFs including a short description and links to your GitHub.

Lastly, I have posted on [r/DearPyGui](https://www.reddit.com/r/DearPyGui/comments/yawihf/deareis_electrochemical_impedance_spectroscopy/). In case you'd like to draw more users towards DearEIS, I'd recommend posting on Reddit in various subreddits.

Thanks again for making a Dear PyGui app that shows off the plotting capabilities!

**Comment by vyrjana** [comment]
Hi,

Thanks for adding DearEIS to the showcase gallery and for spreading the word about the project! The target audience is a rather small demographic but an upcoming article about the project itself (currently undergoing peer review) will hopefully help make the project more widely known in the field.

**Comment by bandit-masked** [comment]
Will you include a citation or reference to Dear PyGui as well?

It would be a nice surprise for the core developers. It's always nice to see open source libraries being used in scientific publications! Dear PyGui has been featured at least [twice](https://github.com/hoffstadt/DearPyGui/wiki/Dear-PyGui-Showcase#scientific-apps-made-with-dear-pygui).

**Comment by vyrjana** [comment]
Yes, the manuscript does include a reference to the Dear PyGui.

**Comment by bandit-masked** [comment]
Could you let us know when it's published?

Are you using the Dear PyGUi Discord server by the way?

**Comment by vyrjana** [comment]
Any preferred way or would it suffice if I added a comment to this thread, which I assume would send you a notification?

Not yet. Is it a lot more active than r/DearPyGui or the discussion section of the repository here on GitHub?

**Comment by bandit-masked** [comment]
A comment to this thread would be just fine as I do receive an e-mail notification when you do.

The Discord server is the primary discussion forum for DPG, much busier than r/dearpygui or GitHub. You seemed to have figured out how to use DPG without it, so I don't know how much value there would be in it for you, yet you are welcome to join.

**Comment by freethebee** [comment]
Hi, your packages looks very interesting. I have been playing a bit over the weekend and it is impressive how much it can do. I will definitely continue using it and suggest it to my colleagues.

To add to the suggestion to post on reddit, there might be interest in r/electrochemistry it is a small subreddit, but it does fit the demographic.

Thanks,
Patrick

**Comment by bandit-masked** [comment]
Thank you for the suggestion! It indeed looks like a small subreddit, but that's fine as it's probably a very relevant community.'

@vyrjana You may be interested in creating a standalone executable at some point. Although slightly more tricky than Pyinstaller, developers have reported excellent results with Nuitka.

**Comment by vyrjana** [comment]
> Hi, your packages looks very interesting. I have been playing a bit over the weekend and it is impressive how much it can do. I will definitely continue using it and suggest it to my colleagues.

Glad to hear it, @freethebee! By the way, if you've installed NumPy from PyPI, then you are (most likely) using the OpenBLAS linear algebra library and as a result the Bayesian Hilbert transform (BHT) method for calculating distributions of relaxation times will be very slow despite utilizing all/most threads to a high degree. I was only able to pinpoint the issue and fix it today. I still have a few other things to implement but the next version will hopefully be ready for release soon.


> @vyrjana You may be interested in creating a standalone executable at some point. Although slightly more tricky than Pyinstaller, developers have reported excellent results with Nuitka.

Perhaps things are better now with the recent 1.0 and 1.1 releases, @bandit-masked. I looked into the possibility of using Nuitka quite a while ago (before 1.0 was available) and at that point it didn't seem particularly viable for me. Compiling a very basic DearPyGui program took a long time and the files that were generated along the way took up a lot of space. Doing that on Windows would be even worse since I only have Windows installed on a T480 (i5-8250U and 8 GB RAM) with very little free space on the SSD.

**Comment by bandit-masked** [comment]
I have read that compiling the code with Nuitka is time consuming. It's the end result that works well, apparently, for the end user. That is, if you want to distribute your application that way. As for file size, maybe setting the correct Nuitka options might help. Compilation will include the Python runtime, regardless of method, so that's 20Mb or so as a baseline size. There are some Python versions made for including distributions, that don't include IDLE and other things. You should probably incorporate those slightly leaner versions.

**Comment by freethebee** [comment]
@vyrjana  indeed, I installed from pypi. I can have a look with conda as well, using mkl numpy.

My experience with nuitka is also a bit hit or miss, although I do like the project a lot, it is no easy task for sure. Perhaps pipx is also an option, although I don't know how convenient that is on windows.

**Comment by vyrjana** [comment]
@freethebee, the latest versions of pyimpspec (3.2.1) and DearEIS (3.3.0) should solve the issue in cases where OpenBLAS or MKL are used.

I already provide an online installer for Windows but it basically just uses pip to install DearEIS and then adds some shortcuts to the start menu.

**Comment by DataExplorerUser** [comment]
@vyrjana Where is the online installer for Windows? I must have missed it on the readme page...

@freethebee pipx works like magic on Windows.

**Comment by vyrjana** [comment]
@DataExplorerUser, it is mentioned in the [Installing](https://github.com/vyrjana/DearEIS#installing) section and also on the [GitHub pages in the step-by-step guide to installing](https://vyrjana.github.io/DearEIS/).

**Comment by DataExplorerUser** [comment]
I must be blind 🤣

**Comment by bandit-masked** [comment]
Congratulations on the publication! First time I learned about The Journal of Open Source Software.

In case you'd like to gather more interesting in your cool project, I'd definitely recommend submitting it to some subreddits. If you share a link to the main subreddit on the Discord server, some people will probably vote you up, which helps the Reddit algorithm and increases exposure to more users on Reddit.

**Comment by vyrjana** [comment]
Thanks! I stumbled upon that journal a while back when I started using [ChemPy](https://github.com/bjodah/chempy).

I may do that but I will hold off for a while longer. I'm currently working on a fairly big refactor/update of [pyimpspec](https://github.com/vyrjana/pyimpspec/) and that will require significant changes to DearEIS as well.

**Comment by bandit-masked** [comment]
Alright, good luck with the development of both DearEIS and pyimpspec!
